src/Tools/ExternalAccess/FSharp/LanguageServices/VisualStudioWorkspaceExtensions.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using Microsoft.VisualStudio.LanguageServices;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.FSharp.LanguageServices
+{
+    public static class VisualStudioWorkspaceExtensions
+    {
+        public static Metadata GetMetadata(this VisualStudioWorkspace workspace, string fullPath, DateTime snapshotTimestamp)
+        {
+            var metadataReferenceProvider = workspace.Services.GetService<VisualStudioMetadataReferenceManager>();
+            return metadataReferenceProvider.GetMetadata(fullPath, snapshotTimestamp);
+        }
+
+        [Obsolete("This is a compatibility shim for F#; please do not use it.")]
```

Comment:
>[Obsolete("This is a compatibility shim for F#; please do not use it.")] [](start = 8, length = 72)

You can remove these Obsolete attributes now.

Reply:
@tmat is correct. Everything in this assembly is by definition an compatibility shim for F#. Nothing else in Roslyn references this assembly, so it is impossible for the shims to be misused. Marking them Obsolete serves no purpose anymore.

Reply:
In this case it does serve a purpose.

We are not supposed to be calling these APIs in VisualFSharp. I'm making them Obsolete to remind myself that they need to go away. Once we have more Roslyn/project-system integration work done, then these APIs will go away.

Reply:
@TIHan that's reasonable. Can you update the message?

```csharp
[Obsolete("Please use [other API] instead.")]
```

Reply:
There technically is no other API to use. The integration would just naturally flow from a `Project` type itself. We simply should not be using this API or an API like this.

Reply:
I'll change the message to: "When Roslyn/ProjectSystem integration finished, don't use this."